### PR TITLE
Add OIDs used by Wi-SUN

### PR DESCRIPTION
**Pull request by MathisMARION**
The Wi-SUN specification mandates the use of `id-on-hardwareModuleName` and `id-kp-wisun-fan-device`.
I tested by displaying this demo certificate: https://raw.githubusercontent.com/SiliconLabs/wisun-br-linux/main/examples/node_cert.pem

Before:
```
$ openssl x509 -in /usr/local/share/doc/wsbrd/examples/node_cert.pem -noout -text
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            3d:1b:6b:15:c0:3a:77:0d:37:b1:3a:43:d5:d0:2b:1e:91:62:eb:98
        Signature Algorithm: ecdsa-with-SHA256
        Issuer: CN = Wi-SUN Demo Root CA
        Validity
            Not Before: Mar  1 07:42:40 2021 GMT
            Not After : Dec 31 23:59:59 9999 GMT
        Subject: CN = Wi-SUN Demo Device
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (256 bit)
                pub:
                    04:a8:ea:ab:ba:78:77:fa:00:2b:f3:76:ba:c2:da:
                    41:f9:08:51:bc:ce:28:f5:d4:dd:55:e3:42:a3:46:
                    a6:5e:1c:db:52:c1:72:97:0e:ce:81:b8:2a:b6:80:
                    8a:42:0b:98:4e:c6:e2:72:97:8b:31:6b:f8:4d:5e:
                    c9:c5:7e:88:8f
                ASN1 OID: prime256v1
                NIST CURVE: P-256
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature, Key Agreement
            X509v3 Extended Key Usage: critical
                1.3.6.1.4.1.45605.1, TLS Web Client Authentication
            X509v3 Subject Alternative Name: critical
                othername: 1.3.6.1.5.5.7.8.4::<unsupported>
            X509v3 Authority Key Identifier: 
                93:64:D4:0E:F7:68:A2:EE:5D:F0:AC:FA:72:BC:C5:1F:4D:4F:3A:24
    Signature Algorithm: ecdsa-with-SHA256
    Signature Value:
        30:45:02:21:00:d0:71:15:63:33:34:c2:b2:03:e9:cc:2b:4b:
        1b:09:4a:6a:2b:58:15:33:27:a8:2a:a8:74:ce:f4:b7:08:ec:
        8b:02:20:31:f2:70:8d:07:b4:64:5b:c4:66:67:45:0c:59:8d:
        ba:83:b3:fa:4d:ba:89:88:02:37:ce:82:a4:4b:1a:49:3e
```
After:
```
$ openssl x509 -in /usr/local/share/doc/wsbrd/examples/node_cert.pem -noout -text
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            3d:1b:6b:15:c0:3a:77:0d:37:b1:3a:43:d5:d0:2b:1e:91:62:eb:98
        Signature Algorithm: ecdsa-with-SHA256
        Issuer: CN=Wi-SUN Demo Root CA
        Validity
            Not Before: Mar  1 07:42:40 2021 GMT
            Not After : Dec 31 23:59:59 9999 GMT
        Subject: CN=Wi-SUN Demo Device
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (256 bit)
                pub:
                    04:a8:ea:ab:ba:78:77:fa:00:2b:f3:76:ba:c2:da:
                    41:f9:08:51:bc:ce:28:f5:d4:dd:55:e3:42:a3:46:
                    a6:5e:1c:db:52:c1:72:97:0e:ce:81:b8:2a:b6:80:
                    8a:42:0b:98:4e:c6:e2:72:97:8b:31:6b:f8:4d:5e:
                    c9:c5:7e:88:8f
                ASN1 OID: prime256v1
                NIST CURVE: P-256
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature, Key Agreement
            X509v3 Extended Key Usage: critical
                Wi-SUN Alliance Field Area Network (FAN), TLS Web Client Authentication
            X509v3 Subject Alternative Name: critical
                othername: Hardware Module Name:<unsupported>
            X509v3 Authority Key Identifier: 
                93:64:D4:0E:F7:68:A2:EE:5D:F0:AC:FA:72:BC:C5:1F:4D:4F:3A:24
    Signature Algorithm: ecdsa-with-SHA256
    Signature Value:
        30:45:02:21:00:d0:71:15:63:33:34:c2:b2:03:e9:cc:2b:4b:
        1b:09:4a:6a:2b:58:15:33:27:a8:2a:a8:74:ce:f4:b7:08:ec:
        8b:02:20:31:f2:70:8d:07:b4:64:5b:c4:66:67:45:0c:59:8d:
        ba:83:b3:fa:4d:ba:89:88:02:37:ce:82:a4:4b:1a:49:3e
```

----

In a future pull request, I would like to add support for displaying the [`HardwareModuleName`](https://datatracker.ietf.org/doc/html/rfc4108#autoid-63) structure.
I have working code, but I am not satisfied with the results, the problem being that the currently supported "general names" all use a single value which is easy to print inline, which is not the case for `HardwareModuleName`.
This is what I have so far, which may be better discussed as part of an other pull request once I open it.
```
            X509v3 Subject Alternative Name: critical
                othername: HwModule:hwType=1.3.6.1.4.1.39873.2,hwSerial=31:32:33:34:35:36
```

### CODE REVIEW

**Comment by MathisMARION** [comment]
Two tests were failing due to the last commit "Remove duplicate colon in otherName display". I fixed the tests, but if this behavior is considered "public API", this commit should be dropped.

**Comment by t8m** [comment]
> Two tests were failing due to the last commit "Remove duplicate colon in otherName display". I fixed the tests, but if this behavior is considered "public API", this commit should be dropped.

That's questionable. IMO it is a bug fix that is OK for master branch. I would not backport such change to stable branches.

**Comment by t8m** [comment]
CI is still relevant.

**Comment by MathisMARION** [comment]
Added OIDs to `fuzz/oids.txt`. (Use `make update` instead of `make generate_crypto_objects` in commit "Regenerate object files".

File: crypto/objects/obj_dat.h [hunk 1]
```diff
@@ -10,7 +10,7 @@
  */
 
 /* Serialized OID's */
-static const unsigned char so[8487] = {
+static const unsigned char so[8504] = {
```
Comment (tom-cosgrove-arm) [inline crypto/objects/obj_dat.h:13]: Hmm, when declared as `so[]` `sizeof(so)` gives `8503` - so we have been off-by-one for a while here. Is the extra byte needed as a null terminator? Or would it be best just to use `... so[] = {`?

Comment (tom-cosgrove-arm) [reply crypto/objects/obj_dat.h:13]: Or is there a reason to size this at one more than the number of bytes?

Comment (tom-cosgrove-arm) [reply crypto/objects/obj_dat.h:13]: (also, `so[]` and `nid_objs[]` are screaming out to be generated by a script, but that's for another time)

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM - agree that the removal of the extra colon should be for `master` only

**Review by t8m — APPROVED** [review]

**Review by tom-cosgrove-arm — APPROVED** [review]
reconfirming

**Comment by t8m** [comment]
Partially squashed and merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**